### PR TITLE
Homework2: Roman Snetkov

### PR DESCRIPTION
**Pull request by romajmg**
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/homework2/task1.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+Calculate the total price in rubles and kopecks for L goods.
+Example:
+Input: The price of one item is 3 rubles 20 kopecks, count 3 items.
+Output: Total price 9 rubles 60 kopecs
+"""
+
+
+# 1st variant
+
+roubles, kopecks, amount = (int(value) for value in input('Input roubles kopecks amount: ').split())
```
Comment (belovmd) [inline src/homework2/task1.py]: please choose one solution and delete redundant code

File: src/homework2/task1.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+sum_roubles = roubles * amount + kopecks * amount // 100
+sum_kopecks = kopecks * amount % 100
+print('Summarized price: ', sum_roubles, 'roubles, ', sum_kopecks, ' kopecks.')
+
+
+# 2nd variant
+
+price = float(input('Input price (e.g. 2.65): '))
```
Comment (belovmd) [inline src/homework2/task1.py]: contradicts task description

File: src/homework2/task2.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+len (list) - the number of items in the list
+"""
+
+import re
+import string
+
+sentence = input('Sentence: ')
+
+
+# Solution 1:
```
Comment (belovmd) [inline src/homework2/task2.py]: please choose one

File: src/homework2/task2.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+sentence1_split = sentence.split()
+for i in range(len(sentence1_split)):
```
Comment (belovmd) [inline src/homework2/task2.py]: range(len(..)) - bad practice

File: src/homework2/task3.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+For example, if "abc cde def" was entered, then "abcdef" should be output.
+"""
+
+
+import string
+
+sentence = input('Input sentence:').replace(' ', '')
+output = ''
+
+for _ in sentence:
```
Comment (belovmd) [inline src/homework2/task3.py]: _ - bad name for variable that is used

File: src/homework2/task4.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+Consider only English letters.
+"""
+
+
+lowercase = 0
+uppercase = 0
+sentence = input('Sentence:')
+
+for _ in sentence:
+    # count upper case letters by code range
```
Comment (belovmd) [inline src/homework2/task4.py]: redundant comments

File: src/homework2/task4.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+Consider only English letters.
+"""
+
+
+lowercase = 0
+uppercase = 0
+sentence = input('Sentence:')
+
+for _ in sentence:
```
Comment (belovmd) [inline src/homework2/task4.py]: _ - bad name

File: src/homework2/task4.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    # count upper case letters by code range
+    if ord(_) in range(65, 91):
```
Comment (belovmd) [inline src/homework2/task4.py]: 'a'<smth<'z'

File: src/homework2/task7.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+If the sides define a triangle, find its area.
+If not, display an invalid data message.
+"""
+
+
+side1, side2, side3 = (int(value) for value in input('Input triangle sides separately: ').split())
+
+if (side1 < side2 + side3) and (side2 < side1 + side3) and (side3 < side1 + side2):
+    p = (side1 + side2 + side3) / 2
+    S = p * (p - side1) * (p - side2) * (p - side3) ** 0.5
```
Comment (belovmd) [inline src/homework2/task7.py]: S - bad naming

Comment (belovmd) [reply src/homework2/task7.py]: this formula doesn't work correctly. you can check with triangle with sides 3, 4, 5 - expected S - 6

**Review by belovmd — CHANGES_REQUESTED** [review]
missed task 8

File: src/homework2/task1.py (hunk 1 above)
Comment (romajmg) [reply src/homework2/task1.py]: Left one solution

File: src/homework2/task1.py (hunk 2 above)
Comment (romajmg) [reply src/homework2/task1.py]: Deleted this solution

File: src/homework2/task2.py (hunk 3 above)
Comment (romajmg) [reply src/homework2/task2.py]: Left one solution

File: src/homework2/task2.py (hunk 4 above)
Comment (romajmg) [reply src/homework2/task2.py]: Changed to "for word in words:"

File: src/homework2/task3.py (hunk 5 above)
Comment (romajmg) [reply src/homework2/task3.py]: Changed variable name to "char"

File: src/homework2/task4.py (hunk 6 above)
Comment (romajmg) [reply src/homework2/task4.py]: Deleted comments

File: src/homework2/task4.py (hunk 7 above)
Comment (romajmg) [reply src/homework2/task4.py]: Changed variable name to "char"

File: src/homework2/task4.py (hunk 8 above)
Comment (romajmg) [reply src/homework2/task4.py]: Reworked

File: src/homework2/task7.py (hunk 9 above)
Comment (romajmg) [reply src/homework2/task7.py]: Changed variable name to "area" and adjusted the formula

**Comment by belovmd** [comment]
9

**Pull request closed without merging**